griptape/tools/computer/tool.py

```diff
@@ -106,18 +107,21 @@ def execute_command_in_container(self, command: str) -> BaseArtifact:
                 detach=True,
             )
 
-            container.wait()
+            if isinstance(container, Container):
+                container.wait()
 
-            stderr = container.logs(stdout=False, stderr=True).decode().strip()
-            stdout = container.logs(stdout=True, stderr=False).decode().strip()
+                stderr = container.logs(stdout=False, stderr=True).decode().strip()
+                stdout = container.logs(stdout=True, stderr=False).decode().strip()
 
-            container.stop()
-            container.remove()
+                container.stop()
+                container.remove()
 
-            if stderr:
-                return ErrorArtifact(stderr)
+                if stderr:
+                    return ErrorArtifact(stderr)
+                else:
+                    return TextArtifact(stdout)
             else:
-                return TextArtifact(stdout)
+                return ErrorArtifact("error running container")
```

Comment:
what are the guidelines for when to return `ErrorArtifact` vs raising an exception? Are ErrorArtifacts specific to tools?

Reply:
`ErrorArtifact`s are not specific to Tools necessarily, but all Tools should return an Artifact of some kind. This information is passed back to the LLM and allows it to potentially course correct.

Beyond Tools, it's kind of situationally dependent on whether a `raise` or `ErrorArtifact` is the correct choice.